libs/vec/native/src/vec/headers/vec.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+#define EXPORT __attribute__((externally_visible,visibility("default")))
```

Comment:
I have a multi-platform EXPORT definition, but this is fine for now - it gives a warning with clang, but we are using GCC anyway. I'll add this in a follow-up PR.